### ISSUE TITLE
Macro filter turned boolean arguments into string

### ISSUE THREAD

**Issue by goooooouwa** [gollum/gollum-lib#446]
Recently I encountered a issue with Naviagation macro, which leads me to the following code in `Gollum::Filter::Macro#extract` method. I noticed that the `extract` method will turn boolean arguments into strings, thus making boolean argument like `false` be treated as `"false"` (a truthy value), making boolean arguments losing the effects.

```ruby
# File 'lib/gollum-lib/filter/macro.rb', line 5

def extract(data)
      # ...
      argstr.scan(/,?\s*(#{arg})\s*/) do |arguments|
      	# Stabstabstab
      	argument = arguments.first
      	
        if argument =~ /^([a-z0-9_]+)="(.*?)"/
      		opts[Regexp.last_match[1]] = Regexp.last_match[2]
			  elsif argument =~ /^"(.*)"$/
				  args << Regexp.last_match[1].gsub("\\\"", "\"")
			  else
				  args << argument
        end
      end
```

## Steps to reproduce the issue:

1. Step 1: define a Navigation macro as `<<Navigation("My TOC", "subdir", false)>>` in a wiki page.
2. Expectation: the generated navigation TOC will not show full path.
3. Actual Result: the generated navigation TOC still shows full path of each page.

Possible root cause:

In `Gollum::Filter::Macro#extract`, the `argstr` variable will be `"\"My TOC\", \"subdir\", false"`, which will be turned into the following `args`: `["My TOC", "subdir", "false"]` (here boolean `false` was turned to string "false"). Then the `args` will be passed to `Gollum::Filter::Macro#process`, it will be treated as a truthy value, which means the `full_path` will never be false, thus never take effects.

**Comment by dometto** [comment]
Thanks for the detailed report! A fix is forthcoming in https://github.com/gollum/gollum-lib/pull/447, which will be included in the 6.0 release.

**Comment by goooooouwa** [comment]
Awesome, that was quick! Thanks for keep making gollum better, love it.

As a workaround for not showing full path for Navigation, what I did for now is simply not passing the boolean argument (so that it will default to false), like this:

`<<Navigation("My TOC","My TOC")>>`
